lib/chef/resource/yum_repository.rb

```diff
@@ -66,11 +66,14 @@ class YumRepository < Chef::Resource
       property :sslclientkey, String, regex: /.*/
       property :sslverify, [TrueClass, FalseClass]
       property :timeout, String, regex: /^\d+$/
-
       property :options, Hash
 
       default_action :create
-      allowed_actions :create, :remove, :make_cache, :add
+      allowed_actions :create, :remove, :make_cache, :add, :delete
+
+      # provide compatibility with the yum cookbook < 3.0 properties
+      alias_method :url, :baseurl
+      alias_method :keyurl, :gpgkey
```

Comment:
For now we're just supporting them until the end of time. If we really care to get everyone on the new properties we can introduce some sort of deprecation logging, but that would add complexity here. 2 lines for backwards compat is simpler.